provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/configs/AttestationMechanism.java

```diff
@@ -66,9 +74,22 @@ else if(attestation instanceof X509Attestation)
             this.x509 = new X509Attestation((X509Attestation)attestation);
             /* SRS_ATTESTATION_MECHANISM_21_008: [If the provided attestation is instance of X509Attestation, the constructor shall set the tpm as null.] */
             this.tpm = null;
+            /* SRS_ATTESTATION_MECHANISM_015: [If the provided attestation is instance of X509Attestation, the constructor shall set the symmetricKey as null.] */
+            this.symmetricKey = null;
             /* SRS_ATTESTATION_MECHANISM_21_007: [If the provided attestation is instance of X509Attestation, the constructor shall set the attestation type as X509.] */
             this.type = AttestationMechanismType.X509;
         }
+        else if(attestation instanceof SymmetricKeyAttestation)
+        {
+            /* SRS_ATTESTATION_MECHANISM_016: [If the provided attestation is instance of SymmetricKeyAttestation, the constructor shall store the provided symmetric keys.] */
+            this.symmetricKey = new SymmetricKeyAttestation((SymmetricKeyAttestation)attestation);
```

Comment:
Can't this just be a cast? 
```java
this.symmetricKey = (SymmetricKeyAttestation) attestation
```

Reply:
Here and above with the other types

Reply:
Then we're better off just casting here. I don't see a need to create a copy

---
In reply to: [250828937](https://github.com/Azure/azure-iot-sdk-java/pull/411#discussion_r250828937) [](ancestors = 250828937)